cedar-drt/fuzz/fuzz_targets/json-then-human-schema-roundtrip.rs

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2022-2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#![no_main]
+use cedar_drt_inner::schemas::semantic_equality_check;
+use cedar_drt_inner::*;
+use cedar_policy_generators::{schema::Schema, settings::ABACSettings};
+use cedar_policy_validator::SchemaFragment;
+use libfuzzer_sys::arbitrary::{self, Arbitrary, Unstructured};
+use serde::Serialize;
+use smol_str::{SmolStr, ToSmolStr};
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Serialize)]
+struct Input {
+    #[serde(skip)]
+    pub schema: SchemaFragment,
+}
+
+/// settings for this fuzz target
+const SETTINGS: ABACSettings = ABACSettings {
+    match_types: false,
+    enable_extensions: true,
+    max_depth: 3,
+    max_width: 7,
+    enable_additional_attributes: false,
+    enable_like: true,
+    // ABAC fuzzing restricts the use of action because it is used to generate
+    // the corpus tests which will be run on Cedar and CedarCLI.
+    // These packages only expose the restricted action behavior.
+    enable_action_groups_and_attrs: false,
+    enable_arbitrary_func_call: true,
+    enable_unknowns: false,
+    enable_action_in_constraints: true,
+    enable_unspecified_apply_spec: true,
+};
+
+impl<'a> Arbitrary<'a> for Input {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        let arb_schema = Schema::arbitrary(SETTINGS.clone(), u)?;
+        let namespace = arb_schema.schema;
+        let name: SmolStr = arb_schema
+            .namespace
+            .map(|name| name.to_smolstr())
+            .unwrap_or_else(|| "".into());
+
+        let schema = SchemaFragment(HashMap::from([(name, namespace)]));
+
+        Ok(Self { schema })
+    }
+
+    fn size_hint(depth: usize) -> (usize, Option<usize>) {
+        Schema::arbitrary_size_hint(depth)
+    }
+}
+
+fuzz_target!(|i: Input| {
+    let json = serde_json::to_value(i.schema.clone()).unwrap();
+    let json_ast = SchemaFragment::from_json_value(json).unwrap();
```

Comment:
I was thinking if it makes sense to decompose the property into two sub-properties: JSON schema round-trip and the one tested by the target above.

Reply:
This also doesn't actually assert anything to check the property that `schema_intermediate_ast -> schema_json_src -> schema_intermediate_ast === id`.